app/src/test/kotlin/com/wire/android/shared/user/email/ValidateEmailUseCaseTest.kt

```diff
@@ -0,0 +1,59 @@
+package com.wire.android.shared.user.email
+
+import com.wire.android.UnitTest
+import kotlinx.coroutines.*
+import kotlinx.coroutines.test.runBlockingTest
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Before
+import org.junit.Test
+import org.mockito.ArgumentMatchers.any
+import org.mockito.Mock
+import org.mockito.Mockito.`when`
+import java.util.regex.Matcher
+import java.util.regex.Pattern
+
+@ExperimentalCoroutinesApi
+class ValidateEmailUseCaseTest : UnitTest() {
+
+    @Mock
+    private lateinit var validateEmailParams: ValidateEmailParams
+
+    private lateinit var validationEmailUseCase: ValidateEmailUseCase
+
+    @Before
+    fun setup() {
+        validationEmailUseCase = ValidateEmailUseCase()
+    }
+
+    @Test
+    fun `Given run is executed, when email doesn't match regex, then return failure`() {
+        `when`(validateEmailParams.email).thenReturn("email")
+
+        runBlockingTest {
+            assertThat(validationEmailUseCase.run(validateEmailParams).isLeft).isTrue()
```

Comment:
I think you have to check also the failure type `EmailInvalid`

Reply:
Good one @mejdoo! Reviewing testing logic is sooo important too. :heart:

Reply:
good point! thanks :)